### PR TITLE
Feature/command line demo tool

### PR DESCRIPTION
**Pull request by Eleonore9**
**Step to build the jar and it's command line options**

Changes:
- create a `-main` function in `run_models.clj` by editing the `output-ccm` function
- add an uberjar profile in the `project.clj`
- create a default configuration file in in the `resources/` directory
- set the three command-line options (two of which have a default value)
- update the `-main` function to parse the command-line options and run the model

_**Note**: changes were made to `fertility_mvp.clj` and `mortality_mvp.clj` while debugging an error that occurred when creating the jar. They end up not being relevant for this particular step, but don't induce any issue and could be used later on._

### CODE REVIEW

File: src/witan/models/run_models.clj [hunk 1]
```diff
@@ -127,11 +129,30 @@
   (core/looping-test (get-datasets inputs gss-code)
                      params))
 
-(defn output-ccm
+(def cli-options
+  [["-i" "--input-config FILEPATH" "Filepath for the config file with inputs info"
+    :default "resources/default_config.edn"]
+   ["-o" "--output-projections FILEPATH" "Filepath for the output projections"
+    :default "resources/ccm_projections.csv"]
```
Comment (acron0) [inline src/witan/models/run_models.clj]: Probably not a good idea to provide a default output, especially one with a directory that mightn't exist

File: src/witan/models/run_models.clj [hunk 2]
```diff
@@ -127,11 +129,30 @@
...
   "runs the ccm, adds district information for easier human
    reading and outputs to a csv"
-  [inputs gss-code params filepath]
-  (println (format "Preparing projection for %s... " (get-district gss-code)))
-  (-> (run-ccm inputs gss-code params)
-      (add-district-to-dataset-per-user-input gss-code)
-      (write-data-to-csv filepath [:gss-code :district :sex :age :year :popn])))
+  [& args]
+  (let [{:keys [input-config output-projections gss-code]}
+        (:options (parse-opts args cli-options))
```
Comment (acron0) [inline src/witan/models/run_models.clj]: What happens if there's an error with the params?

File: src/witan/models/run_models.clj [hunk 3]
```diff
@@ -127,11 +129,30 @@
...
+        input-file (read-string (slurp input-config))
```
Comment (acron0) [inline src/witan/models/run_models.clj]: add try/catch for if file doesn't exist, maybe?

File: src/witan/models/run_models.clj (hunk 1 above)
Comment (Eleonore9) [inline src/witan/models/run_models.clj]: I guess you're right! Although I want the `resources` directory to be in the jar as that's where the default datasets are.

File: src/witan/models/run_models.clj (hunk 3 above)
Comment (Eleonore9) [inline src/witan/models/run_models.clj]: Good idea!

File: src/witan/models/run_models.clj (hunk 1 above)
Comment (seb231) [inline src/witan/models/run_models.clj]: We could replace :default with just a comment saying where an example/default file should be found?

Comment (acron0) [inline src/witan/models/run_models.clj]: @Eleonore9 Right, but you can't write to a directory inside the jar file...

File: src/witan/models/run_models.clj [hunk 4]
```diff
@@ -127,11 +129,38 @@
...
    reading and outputs to a csv"
-  [inputs gss-code params filepath]
-  (println (format "Preparing projection for %s... " (get-district gss-code)))
-  (-> (run-ccm inputs gss-code params)
-      (add-district-to-dataset-per-user-input gss-code)
-      (write-data-to-csv filepath [:gss-code :district :sex :age :year :popn])))
+  [& args]
+  (let [{:keys [input-config output-projections gss-code]}
+        (:options (parse-opts args cli-options))
+        input-file (get-inputs input-config)
```
Comment (Eleonore9) [inline src/witan/models/run_models.clj]: I think we should be able to make it a bit more compact by destructuring like:

``` Clojure
{:keys [input-datasets user-parameters]} (get-inputs input-config)
```

**Comment by otfrom** [comment]
:shipit:

**Pull request merged**